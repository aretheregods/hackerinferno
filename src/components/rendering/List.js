import { Link } from 'inferno-router';
import { Item } from './Item'

function Pager(props) {
    const endpoint_path = props.list;
    const page_number = props.page;
    const more_pages = props.more;
    const forward_path = !page_number ? 2 : page_number + 1;
    return (
        <span id="pager-span" style={{display:"flex","justify-content":"space-around"}}>
            {page_number > 1 ?
                <Link to={`/${endpoint_path}${page_number - 1}`}> {String.fromCharCode(8592)}Back</Link> :
                <a href={`/${endpoint_path}${page_number}`} style={{"pointer-events":"none",cursor:"default",color:"#D3D3D3"}}>{String.fromCharCode(8592)}Back</a>
            }
            {page_number}
            {more_pages ?
                <Link to={`/${endpoint_path}${forward_path}`}>More {String.fromCharCode(8594)}</Link> :
                <a href={`/${endpoint_path}${page_number}`} style={{"pointer-events":"none",cursor:"default",color:"#D3D3D3"}}>More {String.fromCharCode(8594)}</a>
            }
        </span>
    )
}

// The Pure Functional Component Format that all main list pages follow
export function HackerList(props) {

    // Data received from props
    const data = props.component_data
    const page_endpoint = props.component
    const page_number = props.page ? ~~props.page : 1;
    const endpoint_path = page_endpoint !==  'news' ? page_endpoint + '/' : '';


    // Basiic List Item Template
    const list_items = data.map(function(listobject, index) {
        return <Item item={listobject} index={index} page_number={page_number}/>
    });

    //  The List of paginated Stories
    return (
        <div id="hacker-list">
            <div id="items">{list_items}</div>
            <div id="pager">
                <h4>
                    <Pager list={endpoint_path} page={page_number} more={data.length === 30} />
                    {/* <span id="pager-span" style={{display:"flex","justify-content":"space-around"}}>
                        {page_number > 1 ?
                            <Link to={`/${endpoint_path}${page_number - 1}`}> {String.fromCharCode(8592)}Back</Link> :
                            <a href={`/${endpoint_path}${page_number}`} style={{"pointer-events":"none",cursor:"default",color:"#D3D3D3"}}>{String.fromCharCode(8592)}Back</a>
                        }
                        {page_number}
                        {more_pages ?
                            <Link to={`/${endpoint_path}${forward_path}`}>More {String.fromCharCode(8594)}</Link> :
                            <a href={`/${endpoint_path}${page_number}`} style={{"pointer-events":"none",cursor:"default",color:"#D3D3D3"}}>More {String.fromCharCode(8594)}</a>
                        }
                    </span> */}
                </h4>
            </div>
        </div>
    )
}
