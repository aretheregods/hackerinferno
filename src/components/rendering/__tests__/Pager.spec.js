import { render } from 'inferno';
import { renderToSnapshot } from 'inferno-test-utils';
import { Pager } from '../List.js';


it('should render More as active and Back as inactive anchor tags when there are 30 items on page 1', () => {
    const snapshot = renderToSnapshot(
        <Pager list="news/" page={1} length={30} max_length={30}></Pager>
    )

    expect(snapshot).toMatchSnapshot();

})

it('should render both Back and More as active when there are 30 items on page 2', () => {
    const snapshot = renderToSnapshot(
        <Pager list="news/" page={2} length={30} max_length={30}></Pager>
    )

    expect(snapshot).toMatchSnapshot();

})

it('should render both Back and More as inactive anchor tags when there are fewer than 30 items on page 1', () => {
    const snapshot = renderToSnapshot(
        <Pager list="news/" page={1} length={17} max_length={30}></Pager>
    )

    expect(snapshot).toMatchSnapshot();

})

it('should render Back as active and More as inactive anchor tags when there are fewer than 30 items on page 2', () => {
    const snapshot = renderToSnapshot(
        <Pager list="news/" page={2} length={17} max_length={30}></Pager>
    )

    expect(snapshot).toMatchSnapshot()
})
