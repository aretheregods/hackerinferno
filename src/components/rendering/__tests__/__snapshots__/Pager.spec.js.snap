// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Back as active and More as inactive anchor tags when there are fewer than 30 items on page 2 1`] = `
<span
  id="pager-span"
  style={
    Object {
      "display": "flex",
      "justify-content": "space-around",
    }
  }
>
  <a
    href="/news/1"
  >
     
    ←
    Back
  </a>
  <a
    href="/news/2"
    style={
      Object {
        "color": "#D3D3D3",
        "cursor": "default",
        "pointer-events": "none",
      }
    }
  >
    More 
    →
  </a>
</span>
`;

exports[`should render More as active and Back as inactive anchor tags when there are 30 items on page 1 1`] = `
<span
  id="pager-span"
  style={
    Object {
      "display": "flex",
      "justify-content": "space-around",
    }
  }
>
  <a
    href="/news/1"
    style={
      Object {
        "color": "#D3D3D3",
        "cursor": "default",
        "pointer-events": "none",
      }
    }
  >
    ←
    Back
  </a>
  <a
    href="/news/2"
  >
    More 
    →
  </a>
</span>
`;

exports[`should render both Back and More as active when there are 30 items on page 2 1`] = `
<span
  id="pager-span"
  style={
    Object {
      "display": "flex",
      "justify-content": "space-around",
    }
  }
>
  <a
    href="/news/1"
  >
     
    ←
    Back
  </a>
  <a
    href="/news/3"
  >
    More 
    →
  </a>
</span>
`;

exports[`should render both Back and More as inactive anchor tags when there are fewer than 30 items on page 1 1`] = `
<span
  id="pager-span"
  style={
    Object {
      "display": "flex",
      "justify-content": "space-around",
    }
  }
>
  <a
    href="/news/1"
    style={
      Object {
        "color": "#D3D3D3",
        "cursor": "default",
        "pointer-events": "none",
      }
    }
  >
    ←
    Back
  </a>
  <a
    href="/news/1"
    style={
      Object {
        "color": "#D3D3D3",
        "cursor": "default",
        "pointer-events": "none",
      }
    }
  >
    More 
    →
  </a>
</span>
`;
